backend/PolicyAdmin/PolicyConverter.cs

```diff
@@ -0,0 +1,269 @@
+            {
+                foreach (XacmlAttributeAssignmentExpression attributeAssignmentExpression in obligationExpression.AttributeAssignmentExpressions)
+                {
+                    if (attributeAssignmentExpression.Category.AbsoluteUri.Equals(AltinnXacmlConstants.MatchAttributeCategory.MinimumAuthenticationLevel))
+                    {
+                        XacmlAttributeValue astr = attributeAssignmentExpression.Property as XacmlAttributeValue;
+                        policy.RequiredAuthenticationLevelEndUser = astr.Value;
+                    }
+
+                }
+            }
+        }
+
+        private static List<string> GetRuleActions(XacmlAllOf allOf, List<string> action)
+        {
+            foreach (XacmlMatch match in allOf.Matches.Where(m => m.AttributeDesignator.Category.AbsoluteUri.Equals(XacmlConstants.MatchAttributeCategory.Action)))
+            {
+                if (action == null)
+                {
+                    action = new List<string>();
+                }
+
+                action.Add(match.AttributeValue.Value);
+            }
+
+            return action;
+        }
+
+        private static List<string> GetRuleResources(XacmlAllOf allOf, List<string> resource)
+        {
+            foreach (XacmlMatch match in allOf.Matches.Where(m => m.AttributeDesignator.Category.AbsoluteUri.Equals(XacmlConstants.MatchAttributeCategory.Resource)))
+            {
+                if (resource == null)
+                {
+                    resource = new List<string>();
+                }
+
+                resource.Add($"{match.AttributeDesignator.AttributeId.ToString()}:{match.AttributeValue.Value}");
+
+            }
+
+            return resource;
+        }
+
+        private static List<string> GetRuleSubjects(XacmlAllOf allOf, List<string> subject)
+        {
+            foreach (XacmlMatch match in allOf.Matches.Where(m => m.AttributeDesignator.Category.AbsoluteUri.Equals(XacmlConstants.MatchAttributeCategory.Subject)))
+            {
+                if (subject == null)
+                {
+                    subject = new List<string>();
+                }
+
+                subject.Add($"{match.AttributeDesignator.AttributeId.ToString()}:{match.AttributeValue.Value}");
+            }
+
+            return subject;
+        }
+
+        public static XacmlPolicy ConvertPolicy(ResourcePolicy policyInput)
+        {
+
+            XacmlPolicy policyOutput = new XacmlPolicy(new Uri($"{AltinnXacmlConstants.Prefixes.PolicyId}{1}"), new Uri(XacmlConstants.CombiningAlgorithms.RuleDenyOverrides), new XacmlTarget(new List<XacmlAnyOf>()));
+
+            foreach (PolicyRule rule in policyInput.Rules)
+            {
+                policyOutput.Rules.Add(ConvertRule(rule));
+            }
+
+            policyOutput.ObligationExpressions.Add(GetAuthenticationLevelObligation(policyInput.RequiredAuthenticationLevelEndUser));
+
+            return policyOutput;
+        }
+
```

Comment:
Stylecop: double empty lines (several). Is code analysis rules not configured?

Reply:
De har fjernet stylecop. De er ikke så glad i det tydeligvis. De bruker dotnet format. Kjørte det nå uten noen klager. (men jeg har fiket noen av disse double Lines n å